app/lib/github_linkback.rb

```diff
@@ -71,18 +72,22 @@ def is_allowed_project_link?(projects, project)
   def create
     return [] unless SiteSetting.github_linkback_access_token.present?
 
-    links = github_links
-    links.each do |link|
-      next unless [:commit, :pr].include?(link.type)
+    links = []
 
-      if link.type == :commit
-        post_commit(link)
-      elsif link.type == :pr
-        post_pr(link)
-      end
+    DistributedMutex.synchronize("github_linkback_#{@post.id}") do
+      links = github_links
+      links.each do |link|
```

Comment:
Should we combine these two lines?

```suggestion
      github_links.each do |link|
```

Reply:
`links` is returned below.